src/zc/buildout/buildout.py

```diff
@@ -1570,9 +1570,10 @@ def _open(base, filename, seen, dl_options, override, downloaded):
         _dl_options, cache=_dl_options.get('extends-cache'),
         fallback=fallback, hash_name=True)
     is_temp = False
+    cached_filename = None
     if _isurl(filename):
-        path, is_temp = download(filename)
-        fp = open(path)
+        cached_filename, is_temp = download(filename)
+        fp = open(cached_filename)
```

Comment:
I basically just renamed `path` to the more explicit `cached_filename`.
This variable is then used to print the cache's filename in addition to the original one when there's an exception.